src/Fetch.js

```diff
@@ -458,29 +456,39 @@ function startFetch(fetch, successcb, errorcb, progresscb, readystatechangecb) {
 #if FETCH_DEBUG
     console.log('fetch: operation success. e: ' + e);
 #endif
-    if (onsuccess) {{{ makeDynCall('vi', 'onsuccess') }}}(fetch);
-    else if (successcb) successcb(fetch);
+    runtimeKeepalivePop();
+    callUserCallback(function() {
+      if (onsuccess) {{{ makeDynCall('vi', 'onsuccess') }}}(fetch);
+      else if (successcb) successcb(fetch);
+    });
   };
 
   var reportProgress = function(fetch, xhr, e) {
-    if (onprogress) {{{ makeDynCall('vi', 'onprogress') }}}(fetch);
-    else if (progresscb) progresscb(fetch);
+    callUserCallback(function() {
```

Comment:
why is `callUserCallback` needed on a progress event? I don't think it's possible for the runtime to be exited from here.

Reply:
What would expect to happen if you call "exit" or "pthread_exit" from a process event handler then?

Why not just do this everywhere we can?

Reply:
Can you think of any reason why it would be bad to wrap process callbacks in try/catch?  Seems like its good to have single place where we can intercept and handle or log "unwind" and ExitStatus exceptions.

Reply:
Code size is a downside. But I see your point, while it seems illogical to `exit()` in a progress event, a user could do that.